blocks/hero-carousel/hero-carousel.js

```diff
@@ -0,0 +1,91 @@
+import {
+  addEnclosingDiv,
+  createSliderWrapper,
+  createSlider,
+  setSliderIds,
+  createButtonRow,
+  createDottedNav,
+  initSlider,
+} from '../../scripts/lib-carousel.js';
+import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
+
+/**
+ * Optimize thumbnails
+ * @param picture
+ */
+const optimizeThumbnails = (picture) => {
+  picture
+    .querySelectorAll('img')
+    .forEach((img) => {
+      img
+        .closest('picture')
+        .replaceWith(
+          createOptimizedPicture(
+            img.src,
+            img.alt,
+            false,
+            null,
+            null,
+            [{ width: '768' }],
+          ),
+        );
+    });
+};
+
+function parseData(data) {
+  return data.replace(/[\n\s]/g, '');
+}
+
+/**
+ * Get configuration from table containing align property and remove after parsing
+ * @param block
+ * @returns {*[]}: config array literal
+ */
+function getConfig(block) {
```

Comment:
this could be shortened:
- move checkAlign out of function scope
- drop variables that are only used once
- return result directly without intermediate variable

also, suggest to throw error instead, so you can properly catch and process it where the view is controlled.

also, suggest using constant instead of magic number for columns.

```
const NUM_COLUMNS = 3;

...

const checkAlign = (str) => str.includes('left') || str.includes('center') || str.includes('right');

...

function getConfig(block) {
  if (block.children.length === 0 || block.children[0].children.length !== NUM_COLUMNS) {
    throw new Error('Invalid configuration. Table with 3 columns and at least 1 row required');
  }

  return Array.from(block.children).map((slide) => {
    const alignValue = slide.children[1].innerText.replace(/[\n\s]/g, '');
    const align = checkAlign(alignValue) ? alignValue : 'left';
    slide.children[1].remove();
    return { align };
  });
}
```